runtime/j9vm/javanextvmi.cpp

```diff
@@ -464,7 +464,15 @@ JVM_GetClassFileVersion(JNIEnv *env, jclass cls)
 JNIEXPORT void JNICALL
 JVM_VirtualThreadHideFrames(JNIEnv *env, jobject vthread, jboolean hide)
 {
-	/* TODO toggle hiding of stack frames for JVMTI */
+	J9VMThread *currentThread = (J9VMThread *)env;
+	j9object_t threadObject = J9_JNI_UNWRAP_REFERENCE(vthread);
+	j9object_t carrierThread = (j9object_t)J9VMJAVALANGVIRTUALTHREAD_CARRIERTHREAD(currentThread, threadObject);
+	J9VMThread *targetThread = J9VMJAVALANGTHREAD_THREADREF(currentThread, carrierThread);
+	if (JNI_TRUE == hide) {
+		targetThread->privateFlags |= J9_PRIVATE_FLAGS_VIRTUAL_THREAD_HIDDEN_FRAMES;
```

Comment:
This code is invalid - privateFlags are only allowed to be modified by the thread that owns them unless under very specific circumstances (like holding exclusive VM access). The flag will need to be somewhere else, and protected by a monitor.

Reply:
The flag has to be applied to the vthread, not the J9VMThread (all virtual thread info must be stored in the vthread since they move between J9VMThreads).